contents/handbook/engineering/on-call-rotation.md

```diff
@@ -0,0 +1,49 @@
+
+To be ready, make sure you have access to:
+
+- PostHog Cloud admin interfaces ([🇺🇸](https://app.posthog.com/admin/)  / [🇪🇺](https://eu.posthog.com/admin/)) - these require another team member to mark you as a staff user in both regions
+- Metabase ([🇺🇸](http://metabase-prod-us/)  / [🇪🇺](http://metabase-prod-eu/)) – these might require another team member to invite you to both regions
+- Our [tailnet](https://github.com/PostHog/posthog-cloud-infra/blob/main/terraform/environments/README.md#connect-to-a-service-hosted-in-our-internal-network), which gates our internal services (such as Grafana and runbooks)
+- [EKS](http://runbooks/eks/) over `kubectl` / `k9s`, in case you need to run Kubernetes cluster operations (such as restarting a pod)
+- [pganalyze](https://app.pganalyze.com/organizations/posthog/)
+
+A chunk of our high-frequency alerts have runbooks attached, which live in our internal runbooks site: [http://runbooks/](http://runbooks/) (also accessible as [go/rb](http://go/rb/)). This site is part of our tailnet, as mentioned above.
+
+When an alert fires, find if there's a runbook for it. A runbook tells you what to look at and what fixes exist. Every alert also has a link to the Grafana graph that triggered the alert.
+
+[If the alert is starting to have any noticeable impact on users, go raise an incident.](https://posthog.com/handbook/engineering/incidents) It's that simple.
+
+If you're stumped and no resource is of help, get someone from the relevant team to shadow you while you sort the problem out. The idea is that they can help you understand the issue and where to find how to debug it. The idea is _not_ for them to take over at this point, as otherwise you won't be able to learn from this incident.
+
+## Why is the on-call rotation spread across all engineers?
+
+If you're in a product team, it's tempting to think that service alerts don't apply to you, or that when you're on call you can just hand everything off to the infrastructure team. That's not the case, because it's important that every engineer has a basic understanding of how our software is deployed, where the weak points in our systems are, and what the failure modes look like. This understanding should be all that's needed to follow the runbooks, and if you follow the causes of alerts, ultimately you'll be less likely to ship code that takes PostHog down.
+
+Besides knowledge, being on call requires availability – including weekends. If teams had their own separate rotations, there would be more people on call in total, and each would have to stand by 24/7 as our teams aren't big enough to follow the sun. This would be more stressful because of availability constraints, while being less productive because of the rare alerts being spread across multiple people.
```

Comment:
> including weekends

Just to be sure, this is only if you get paged, you're not expected to be online during the whole weekend monitoring the systems, right? can we be more clear about that?

Reply:
To me that felt covered by the "what do you do on call" part
> You just go about your day, paying no attention to being on call most of the time! But keep your laptop at hand, because when you do get paged, it's time to switch gears immediately and start investigating the alert.

Reply:
Right, maybe because I just reviewed/read the support hero handbook and it was explicit about the Mondays-Fridays, I assumed that here too, and only at the end of the page did I read the "including weekends", probably an interpretation on my end, happy to keep as it is if this is clear for others as well.

Reply:
Good point! Reading this again I agree it's not clear what the deal with weekends is. Added this:
> Because the stability of production systems is critical, on-call involves weekends too (unlike Support Hero). More likely than not, nothing will happen over the weekend – but you never know, so the important thing is to keep your laptop at hand.